backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlTableMetadataExecutor.java

```diff
@@ -396,4 +395,52 @@ private static void executeEnableSearch(DSLContext jooq, TableMetadata table) {
 
     createSearchTrigger(jooq, table, table.getTableName());
   }
+
+  static void checkNoColumnWithSameNameExistsInSubclass(
```

Comment:
would have expected a model check instead of check in db , looks oke , but large complex query like this is hard to review / maintain / extend / fix . My not use the (java)model for this check

Reply:
I initially used the Java api however that turned out to be slow and could become impossible slow, because it requires to iteratively retrieve ALL schemas in the database (which might many!!!) and then iterate through all inherited tables (which it cannot know from the parent schema). This query queries the metadata model as it is stored in the database so we can make use of postgresql filter capability.